### PR TITLE
Build baru for wasm32-unknown-unknown

### LINKED ISSUE #26: Ensure it works in WASM on all platforms
1. We should ensure the library can be compiled to WASM
2. Compilation to WASM has to work on all platforms we use for development (Linux and MacOS)

### PR DESCRIPTION
**Pull request by luckysori**
Fixes #26.

### CODE REVIEW

**Review by rishflab — APPROVED** [review]

**Review by thomaseizinger — APPROVED** [review]
Tests are only running on `ubuntu` at the moment (line 58). That is because we cannot run testcontainer based tests on MacOS.

Hence, this entire script might be cleaner if you change the `build_test_workspace` to be just `build` but extend the matrix to build on wasm as well.

Then you can add one more job that just runs on ubuntu and runs the tests. That should get the number of conditionals in your job definition down.

**Comment by luckysori** [comment]
> Tests are only running on `ubuntu` at the moment (line 58). That is because we cannot run testcontainer based tests on MacOS.

Maybe we can run tests on MacOS after/if we merge #34. Let's put this on hold since it didn't make it to the 0.2.0 release anyway.

**Comment by luckysori** [comment]
Closing so that I don't run CI both on my fork and here.

**Pull request closed without merging**
